screens/settings/NameChangeScreen.js

```diff
@@ -0,0 +1,161 @@
+import { FontAwesome5 } from '@expo/vector-icons';
+import PropTypes from 'prop-types';
+import React from 'react';
+import { AsyncStorage, View } from 'react-native';
+import AuthTextField from '../../components/AuthTextField';
+import {
+  BigTitle,
+  ButtonLabel,
+  FilledButtonContainer,
+  Subhead,
+} from '../../components/BaseComponents';
+import Colors from '../../constants/Colors';
+import { getCustomersById, updateCustomers } from '../../lib/airtable/request';
+import { inputFields } from '../../lib/authUtils';
+import {
+  AuthScreenContainer,
+  BackButton,
+  FormContainer,
+} from '../../styled/auth';
+
+export default class NameChangeScreen extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      customer: null,
+      success: false,
+      values: {
+        [inputFields.NAME]: '',
+      },
+      errors: {
+        [inputFields.NAME]: '',
+      },
+    };
+  }
+
+  // Load customer record
+  async componentDidMount() {
+    const customerId = await AsyncStorage.getItem('customerId');
+    try {
+      const customer = await getCustomersById(customerId);
+
+      this.setState({ customer });
+    } catch (err) {
+      console.error(err);
+    }
+  }
+
+  // Check for an error with updated text
+  // Set errors and updated text in state
+  updateError = async (text, inputField) => {
+    let error = false;
+    let errorMsg = '';
+    // validate returns null if no error is found
+    switch (inputField) {
+      case inputFields.NAME:
+        error = !text.replace(/\s/g, '').length;
+        if (error) errorMsg = 'Name cannot be blank';
+        break;
+      default:
+        console.log('Not reached');
+    }
+    this.setState((prevState) => ({
+      errors: { ...prevState.errors, [inputField]: errorMsg },
+      values: { ...prevState.values, [inputField]: text },
+    }));
+
+    return error;
+  };
+
+  // onBlur callback is required in case customer taps on field, does nothing, and taps out
+  onBlur = async (inputField) => {
+    await this.updateError(inputField);
+  };
+
+  // onTextChange does a check before updating errors
+  // It can only remove errors, not trigger them
+  onTextChange = async (text, inputField) => {
+    // Only update error if there is currently an error
+    if (this.state.errors[inputField]) {
+      await this.updateError(text, inputField);
+    } else {
+      this.setState((prevState) => ({
+        values: { ...prevState.values, [inputField]: text },
+      }));
+    }
+  };
+
+  changeName = async () => {
```

Comment:
The reload method I mentioned in Slack:
I added `Updates.reload();` from `import { Updates } from 'expo';` here. 
If we modify the designs to warn users that it has to refresh for changes to take effect, this reload would probably need to happen in the confirmation stage instead. But also, if we have it there, the back button could get confusing, since it might imply that the change doesn't happen until you do the refresh, so if they go back, they might expect it to not change (and it visually won't change but it would in Airtable)... so maybe we could also move the `updateCustomers` to the confirmation too and make that the final step?? Sorry let me know if that makes no sense to you.

Reply:
Note: this requires using Updates from Expo SDK36 because the SDK37 version with `expo-updates` module gave me issues. It would perfectly fine in development, but when I tried it in production with expo publish, it gave me this error `ENOENT: no such file or directory, open '/Users/anniewang/Development/dccentralkitchen/android/app/src/main/AndroidManifest.xml'`. If you want to try the new `expo-updates` version, you can use this in the Expo dev tools window.
![image](https://user-images.githubusercontent.com/21160510/81440658-43785b00-9125-11ea-8482-7d25e43029d8.png)

Reply:
my take: get rid of the back button on the success page, have button say 'click here to refresh' and some text above changed to 'refresh to see changes'